### ISSUE TITLE
Galerie-Element "vergißt" ausgewählte Bilder

### ISSUE THREAD

**Issue by Samson1964** [contao/core-bundle#1487]
Kann in der Demo nachgestellt werden:

1. Galerie-Element anlegen und Bilder aus einem Unterordner wählen. Dabei den Unterordner im Pfad fest einstellen.
2. Dateimanager aufrufen und einen anderen Pfad dort einstellen.
3. Zurück zum Galerie-Element wechseln und weitere Bilder aus dem anderen Pfad hinzufügen. Nach dem Klick auf Anwenden sind die Bilder aus dem Pfad in Punkt 1 verschwunden.

**Comment by Samson1964** [comment]
Ja das meine ich. In 3.5 funktioniert das noch wunderbar. Dort wird der eingestellte Pfad ignoriert und die bereits ausgewählten Bilder sind mit zu sehen. C4 zeigt dagegen nur den eingestellten Pfad und vergißt die anderen Bilder.

**Comment by xchs** [comment]
> Ein ähnliches Ticket gab es erst vor kurzem - finde es aber gerade nicht.

https://github.com/contao/core-bundle/issues/1449 ?

**Comment by leofeyer** [comment]
Contao kann immer nur die Checkboxen berücksichtigen, die im Formular vorhanden sind. Wenn Du nur einen Teilbaum anzeigst, können also auch nur Bilder aus diesem Teilbaum ausgewählt werden.

Ich weiß nicht ob oder wie das in Contao 3 anders sein kann, aber wenn ist es wohl eher nur zufällig so.

**Comment by fritzmg** [comment]
> contao/core-bundle#1449 ?

Genau, danke - aber da hat sich ja herausgestellt, dass das zuklappen die Elemente darin nicht deselektiert sondern tatsächlich ein anderer Bug vorlag.

**Comment by Samson1964** [comment]
Contao 4.4:
Sind die Bilder in Unterverzeichnissen des fest eingestellten Pfades, dann bekommen die Bilder auch ihre Checkboxen und das Verzeichnis mit den Bildern wird automatisch mit aufgeklappt, damit man die ausgewählten Bilder sieht. 
Sind die Bilder NICHT in Unterverzeichnissen des fest eingestellten Pfades, dann werden die Bilder komplett ignoriert. Klickt man dann versehentlich auf Anwenden, sind die Bilder "weg".

Contao 3.5:
... ignoriert bei der Bildauswahl den fest eingestellten Pfad. Da kann man einstellen in der Dateiverwaltung was man will: C3 öffnet immer den zuletzt angezeigten Baum im Galerie-Element (also den hier zuletzt eingestellten Baum) mit allen relevanten Checkboxen.

Wenn das Verhalten in C3 zufällig so war, dann war es zufällig richtig. Wenn es in C4 auch ein Zufall ist, dann ist es diesmal aber ein unerwünschter Zufall. Welcher Benutzer würde denn beim Klick auf "Dateien auswählen" nicht erwarten, das die bereits ausgewählten Dateien eine aktivierte Checkbox bekommen?!

**Comment by leofeyer** [comment]
> Contao 3.5:
> ... ignoriert bei der Bildauswahl den fest eingestellten Pfad. Da kann man einstellen in der Dateiverwaltung was man will:

An dieser Beschreibung alleine sieht man ja schon, dass es nicht korrekt ist. Wenn ich einen Pfad vorgebe, möchte ich auch, dass er berücksichtigt wird. Wozu sonst die Möglichkeit?

**Comment by Samson1964** [comment]
Aber das die ausgewählten Bilder verloren gehen, wenn man z.B. weitere Bilder hinzufügen möchte, ist bestimmt auch nicht beabsichtigt.

**Comment by asaage** [comment]
Ich glaube, es sollte nochmal geklärt werden was damit gemeint ist

> Wenn ich einen Pfad vorgebe

> den Unterordner im Pfad fest einstellen

geht's um dca (eval path)?
geht's darum, dass der Redakteur evtl. keinen Zugriff auf den aktuell benutzen Pfad hat?
geht's darum, dass der Redakteur zuvor im Filemanager einfach einen anderen Ast ausgewählt hatte?

**Comment by asaage** [comment]
Habe ich auch... das hat ja aber nicht's mit **fest eingestellt** zu tun sondern ist nur der letzte Status des Filemanagers.

In dem Fall würde ich behaupten, der eigentliche Bug liegt darin, dass der Filemanager nicht dorthin springt, wo bereits Bilder ausgewählt waren. 😄

**Comment by Samson1964** [comment]
Genau so ist es :-)

**Comment by leofeyer** [comment]
Wie am 17. Mai auf Mumble besprochen, wäre eine Meldung unterhalb des Dateimanager-Menüs sinnvoll, die auf weitere ausgewählte Dateien hinweist und eine Checkbox anbietet, um diese abzuwählen.

Ob das jedoch ein Feature ist, dass viele wirklich brauchen? Wenn ja, bitte kommentiert doch kurz hier. 😉

**Comment by fritzmg** [comment]
Vor kurzem ist mir das in einem anderen Kontext negativ aufgefallen: in einem "Individuelle Navigation" Modul wollte ich eine zusätzliche Seite hinzufügen. Da der Seitenbaum schon relativ groß ist, habe ich die gesuchte Seite einfach in der Suche des Seitenbaum Wählers eingegeben und dann ausgewählt.

Da diese Filterung natürlich alle anderen Seiten ausschließt, waren dann alle bisherigen Seiten natürlich abgewählt und es blieb nur die gesuchte Seite ausgewählt. Mir blieb also nichts anderes über als _keine_ Filterung (weder Suche noch Subtree) zu verwenden und die gesuchte Seite statt dessen über `Ctrl+F` im Browser zu finden.

**Comment by asaage** [comment]
Ich denke, wenn die Auswahl nicht leer ist (wie bei einem neuen Element)
sollte der Filter bereits ausgewählte Elemente berücksichtigen (also auf den kleinsten gemeinsamen parent wechseln)
Vielleicht geht das ja sogar temporär.

**Comment by dmolineus** [comment]
Dass die Bilder deselektiert werden, die nicht sichtbar sind, tritt auch auf wenn man die Suche betätigt. Bei größeren Bildersammlungen schon Recht ärgerlich. Also bin daher für das Feature.

**Comment by Samson1964** [comment]
Im Übrigen tritt das nicht nur beim Galerie-Element auf, sondern wie schon @fritzmg angemerkt hat, auch bei anderen Auswahldialogen. Wie z.B. beim Downloads-Element.

**Comment by Aybee** [comment]
Ich kann mich @fritzmg da auch nur anschließen. Mir ist dies vor ein paar Tagen auch passiert, da ich nicht damit gerechnet hatte, dass bereits Ausgewähltes verschwindet, wenn man den Filter einsetzt.

**Comment by ngdot** [comment]
Ich push das Thema mal! Bin definitiv für das Feature.

**Comment by MacKP** [comment]
Ich kann mich hier auch nur noch mal anschließen. Das die Dateiauswahl die Ausgewählten Bilder vergisst, wenn man Filtert o.ä. ist bei größeren Installationen echt unschön. Und leider auch sehr schwer dem Kunden zu vermitteln ;-) Mit dieser Limitierung macht das Suchen/Filtern irgendwie nicht mehr so richtig Sinn ^^
